### ISSUE TITLE
Spikes in data

### ISSUE THREAD

**Issue by jiyi-deepfield** [facebook/prophet#1013]
Hi! I am using fbprophet on predicting data that come with every 5 minutes. One of the data sets has very high spikes in it. When plotted it looks like below.
![Singapore](https://user-images.githubusercontent.com/50632078/59538349-6ffd0600-8ec7-11e9-952f-655b9fc05fbc.png)
The entire dataset is given below.
[sample_data.txt](https://github.com/facebook/prophet/files/3292237/sample_data.txt)

Is there a good way to incorporate the spikes into the prediction? I want the spikes because the spikes are only showing at the same time every day. I have tried holidays but it doesn't work. I wonder if it is because fbprophet is not designed to analyze time series with only 5min apart.
Any suggestion would be appreciated.
Edit: I have noticed that I usually have a spike at 02:30:00 every day. I tried to make this time a holiday.
Basically I implemented the following code
`spike_time = list(pd.date_range(start='2019-05-18 02:30:00', end='2019-06-05 02:30:00', freq='D'))`
`spike_time = [e.strftime('%Y-%m-%d %H:%M:%S') for e in spike_time]`
`spikes = pd.DataFrame({
  'holiday': 'spikes',
  'ds': pd.to_datetime(spike_time),
  'lower_window': 0,
  'upper_window': 0,
})`
`m = Prophet(holidays=spikes)`
`m.fit(data)`
`future = m.make_future_dataframe(periods=0, freq='5min')`
`fcst = m.predict(future)`
`fig = m.plot_components(fcst)`
The plot I am getting looks like below
![image](https://user-images.githubusercontent.com/50632078/59623784-18e37500-9103-11e9-971e-ae7720a0f377.png)
which does not look great

**Comment by bletham** [comment]
There's no issue necessarily with having 5-min interval data (the underlying model is continuous), but by default the model will do a bad job on this time series because the model assumes smooth seasonality. That assumption comes because seasonality is modeled with a truncated Fourier series, which basically means it cannot change very rapidly. In this case you can see that the daily seasonality is jumping up to try and capture some of the spike, but the spike is very short and the daily seasonality cannot change quickly enough to grab it, so it only grabs a little of it.

The solution is indeed to model that specific effect. With daily data adding a holiday would be exactly what you want to do, but here it didn't work because holidays operate at the level of a day, and so the effect is applied to an entire day - not just the 5 or so mins that you want.

What you instead need to do to capture this effect is to use a binary extra regressor. Add a column to your dataframe that is "is_0230" that is 1 if the ds of that row is 02:30, and 0 otherwise. Add the same column to your future dataframe. The model will then be able to learn this spike, and will add it to the future anytime "is_0230" is set to 1. See the docs here for more about adding an extra regressor: https://facebook.github.io/prophet/docs/seasonality,_holiday_effects,_and_regressors.html#additional-regressors

**Comment by jiyi-deepfield** [comment]
Thank you! This is extremely helpful.
